addons/website/static/src/components/dialog/dialog.js

```diff
@@ -124,57 +116,406 @@ export class AddPageDialog extends Component {
+            }
+            mainEl.appendChild(wrapEl);
+            await wUtils.onceAllImagesLoaded($(wrapEl));
+            // Restore image lazy loading.
+            for (const imgEl of lazyLoadedImgEls) {
+                imgEl.setAttribute("loading", "lazy");
+            }
+            // Wait for fonts.
+            await iframeEl.contentDocument.fonts.ready;
+            holderEl.classList.remove("o_loading");
+            const adjustHeight = () => {
+                if (!this.previewRef.el) {
+                    // Stop ajusting height when preview is removed.
+                    return;
+                }
+                const outerWidth = parseInt(window.getComputedStyle(previewEl).width);
+                const innerHeight = wrapEl.getBoundingClientRect().height;
+                const innerWidth = wrapEl.getBoundingClientRect().width;
+                const ratio = outerWidth / innerWidth;
+                iframeEl.height = Math.round(innerHeight);
+                previewEl.style.setProperty("height", `${Math.round(innerHeight * ratio)}px`);
+                // Sometimes the final height is not ready yet.
+                setTimeout(adjustHeight, 50);
+                holderEl.classList.add("o_ready");
+            };
+            adjustHeight();
+        });
+    }
+
+    select() {
+        if (this.holderRef.el.classList.contains("o_loading")) {
+            return;
+        }
+        if (this.props.isBlank) {
+            this.props.addPage();
+            return;
+        }
+        const wrapEl = this.iframeRef.el.contentDocument.getElementById("wrap").cloneNode(true);
+        for (const previewEl of wrapEl.querySelectorAll(".o_new_page_snippet_preview")) {
+            previewEl.remove();
+        }
+        this.props.addPage(wrapEl.innerHTML);
+    }
+}
+AddPageTemplatePreview.props = {
+    addPage: Function,
+    getCssLinkEls: Function,
+    template: Object,
+    animationDelay: Number,
+    firstRow: {
+        type: Boolean,
+        optional: true,
+    },
+    isBlank: {
         type: Boolean,
         optional: true,
     },
+    isPreloading: {
+        type: Boolean,
+        optional: true,
+    },
+};
+AddPageTemplatePreview.template = "website.AddPageTemplatePreview";
+
+export class AddPageTemplatePreviews extends Component {
+    setup() {
+        super.setup();
+    }
+
+    get columns() {
+        const result = [[], [], []];
+        let currentColumnIndex = 0;
+        // Do not touch props when reordering templates.
+        const templates = Array.from(this.props.templates);
+        if (templates[0].key !== 'website.new_page_template_sections_basic_blank') {
+            const blankTemplate = templates.find(template => template.key === 'website.new_page_template_sections_basic_blank');
+            if (blankTemplate) {
+                // Put blank as first template.
+                templates.pop(blankTemplate);
+                templates.unshift(blankTemplate);
+            }
+        }
```

Comment:
In another comment I was suggesting to remove the "basic blank" preview from the regular preview component and add it in JS with its own onclick bound at that time.

Wouldn't here be a good fit?
It would also allow removing this "reordering" code on top of the rest